### PR TITLE
Add support for sprite slicing

### PR DESCRIPTION
**Pull request by iamOgunyinka**
This PR adds support for slicing. Due to the nature of the preference, the default slice color, if any, defaults to blue.

### CODE REVIEW

File: src/app/file/psd_format.cpp [hunk 1]
```diff
@@ -364,16 +367,31 @@ class PsdDecoderDelegate : public psd::DecoderDelegate {
       const int timeMs = frameInfo.duration * 10;
       m_sprite->setFrameDuration(frame_t(i), timeMs);
     }
-    // at this point, we're supposed to be able to set the activeFrame
+
+    auto& slices = m_sprite->slices();
+    for (const auto& slice : m_slices.slices) {
```
Comment (dacap) [inline src/app/file/psd_format.cpp:372]: Couldn't we do this just in `onSlicesData()` without storing the slices in `m_slices`?

**Comment by dacap** [comment]
I've just tested and it looks like the cel is not correctly aligned, and "disabled" slices shouldn't be imported:

Screen Shot 2021-12-01 at 09 57 53

Result:

Screen Shot 2021-12-01 at 09 57 41

Expected:

Screen Shot 2021-12-01 at 09 59 33

**Comment by iamOgunyinka** [comment]
@dacap thanks for your review, I'll take another look at this.

**Comment by dacap** [comment]
Attaching the slices.psd file inside a zip: [slices.psd.zip](https://github.com/aseprite/aseprite/files/7641554/slices.psd.zip)

**Pull request closed without merging**